pkg/apis/networking/types.go

```diff
@@ -239,9 +239,19 @@ type IngressSpec struct {
 	// rule. At least one of 'backend' or 'rules' must be specified. This field
 	// is optional to allow the loadbalancer controller or defaulting logic to
 	// specify a global default.
+	// Deprecated 1.16
 	// +optional
 	Backend *IngressBackend
 
+	// The default backend capable of servicing requests that don't match any
+	// rule. At least one of 'defaultBackend' or 'rules' must be specified. This field
+	// is optional to allow the loadbalancer controller or defaulting logic to
+	// specify a global default. This type is a rename for the spec.Backend for clarity
+	// and will replace the Backend type entirely in a future release.
+	// This type is GA for 1.16
+	// +optional
+	DefaultBackend *IngressBackend
```

Comment:
the internal type should only have a single field. the `backend` field in v1beta1 and the `defaultBackend` field in v1 should both convert to/from the same internal field

Reply:
if you name the internal field to `DefaultBackend`, you will need to write v1beta1 custom conversion functions to convert the v1beta1 `Backend` field to/from that field.

if you name the internal field to `Backend`, you will need to write v1 custom conversion functions to convert the v1 `DefaultBackend` field to/from that field.

Reply:
That makes more sense, thanks a ton @liggitt